py3status/modules/wwan.py

```diff
@@ -342,38 +343,80 @@ def _get_bearer(self, modem_proxy):
-            sms_data = {}
+        if messages:
+            sms_data = []
             for msg in messages:
                 id = int(msg.replace(STRING_MODEMMANAGER_SMS_PATH, ''))
                 sms_proxy = self.bus.get(STRING_MODEMMANAGER_DBUS, msg)
-                contact = sms_proxy.SMSC
+                contact = sms_proxy.Number
                 text = sms_proxy.Text
-                key += 1
-                sms_data[key] = {'id': id, 'contact': contact, 'text': text}
+                text_short = self._truncate_message(text)
                 messages_data['last_contact'] = '{}'.format(contact)
                 messages_data['last_text'] = '{}'.format(text)
                 messages_data['last_text_short'] = '{}'.format(
-                    self._trunc_message(text))
-                messages_data['last_message'] = '{}: {}'.format(contact, text)
-                messages_data['last_message_short'] = '{}: {}'.format(
-                    contact, self._trunc_message(text))
+                    self._truncate_message(text))
+                sms_data.insert(
+                    0, {
+                        'id': id,
+                        'contact': contact,
+                        'text': text,
+                        'text_short': text_short
+                    })
             messages_data['messages'] = sms_data
 
-            # get sms counter
-            messages_counter = len(messages_data['messages'])
-            messages_data['counter'] = messages_counter
+            # get last message infos
+            messages_data['last_message'] = '{}: {}'.format(
+                sms_data[-1]['contact'], sms_data[-1]['text'])
+            messages_data['last_message_short'] = '{}: {}'.format(
+                sms_data[-1]['contact'], sms_data[-1]['text_short'])
+        return messages_data
+
+    def _get_messages(self, modem_proxy):
+        messages = {}
+        messages_data = {}
+        new_messages = False
+
+        # get messages from modem
+        try:
+            messages = modem_proxy.Messages
         except:
             pass
 
+        messages_data = self._organize_messages(messages)
+
+        if self.py3.storage_get('wwan_messages_counter') == None:
+            previous_messages_counter = 0
+        else:
+            previous_messages_counter = int(
+                self.py3.storage_get('wwan_messages_counter'))
+
+        # get sms counter
+        messages_counter = len(messages_data['messages'])
+        messages_data['previous_messages_counter'] = previous_messages_counter
+        messages_data['messages_counter'] = messages_counter
+
         return messages_data
 
-    def _trunc_message(self, message):
+    def _notify_message(self, messages_data):
+        new_messages = False
+        # set new message placeholder to true
+        if self.notify_on_new_message == True and messages_data['messages_counter'] > messages_data['previous_messages_counter']:
+            new_messages = True
```

Comment:
We can skip new config `notify_on_new_message = False` by using `allow_urgent = False` instead.

Reply:
Sorry for not clearing this up. We already can do this. Do this instead.
```
if messages_data['counter'] > messages_data['previous_messages_counter']:
```